repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -973,6 +976,18 @@ public void removeBusinessAttributes(AtlasVertex entityVertex, AtlasEntityType e
         }
     }
 
+    public static void validateProductStatus(AtlasVertex assetVertex) throws AtlasBaseException {
+        if (assetVertex != null) {
+            if (DATA_PRODUCT_ENTITY_TYPE.equals(assetVertex.getProperty(TYPE_NAME_PROPERTY_KEY, String.class))) {
+                String entityState = assetVertex.getProperty(STATE_PROPERTY_KEY, String.class);
+
+                if ((AtlasEntity.Status.DELETED.name().equals(entityState))) {
+                    throw new AtlasBaseException(OPERATION_NOT_SUPPORTED, "Cannot update DataProduct that is Archived!");
+                }
+            }
+        }
+    }
+
```

Comment:
I see duplication of method in repository/src/main/java/org/apache/atlas/repository/store/graph/v2/ClassificationAssociator.java

Avoid duplication